src/modules/layoutUiRules/rulesHelper.js

```diff
@@ -38,3 +38,25 @@ export const getAxisPerLockedDimByVisType = (visType, dimensionId) =>
 
 export const getAllLockedDimIdsByVisType = visType =>
     Object.keys(getLockedDimsByVisType(visType))
+
+export const isDimensionLocked = (visType, dimensionId) =>
+    getAllLockedDimIdsByVisType(visType).find(id => id === dimensionId)
+
+export const isAxisFull = (visType, axisId, axisDimensionsCount) =>
+    axisDimensionsCount === getAxisMaxNumberOfDimsByVisType(visType, axisId)
```

Comment:
should check >= instead of === ?

Reply:
I guess it won't hurt

Reply:
Having the count greater than the max limit indicates that somewhere the rule has been violated.
Could that perhaps happen when loading an existing visualization?

Reply:
Not sure when/if this would happen, but for future proofing I would check `>=`, since greater than (even though there might be an error somewhere) still counts as "full" right?

Reply:
Fixed.